polling/src/main/java/io/atleon/polling/AloPollingReceiver.java

```diff
@@ -0,0 +1,64 @@
+package io.atleon.polling;
+
+import io.atleon.core.Alo;
+import io.atleon.core.AloFactory;
+import io.atleon.core.AloFlux;
+import io.atleon.core.OrderManagingAcknowledgementOperator;
+import io.atleon.polling.reactive.Pollable;
+import io.atleon.polling.reactive.PollerOptions;
+import io.atleon.polling.reactive.PollingReceiver;
+import io.atleon.polling.reactive.PolledRecord;
+import io.atleon.polling.reactive.PollingSourceConfig;
+import io.atleon.polling.reactive.ReceiverRecord;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+public class AloPollingReceiver<P, O> {
+
+    private final Pollable<P, O> pollable;
+    private final PollingSourceConfig config;
+
+    private AloPollingReceiver(final Pollable<P, O> pollable,
+                               final PollingSourceConfig config) {
+        this.pollable = pollable;
+        this.config = config;
+    }
+
+    public static <P, O> AloPollingReceiver<P, O> from(final Pollable<P, O> pollable,
+                                                       final PollingSourceConfig config) {
+        return new AloPollingReceiver<>(pollable, config);
+    }
+
+    public AloFlux<P> receiveRecords() {
+        return Flux.defer(() -> receiveAloRecords()
+                        .map(PolledRecord::getPayload))
+                .as(AloFlux::wrap);
+    }
+
+    private AloFlux<PolledRecord<P, O>> receiveAloRecords() {
+        final AloFactory<PolledRecord<P, O>> aloFactory = new AloPollingRecordFactory<>();
+        return PollingReceiver.create(pollable, buildPollerOptions()).receive()
+                .transform(r -> createAloRecords(r, aloFactory))
+                .as(AloFlux::wrap);
+    }
+
+    private PollerOptions buildPollerOptions() {
+        return PollerOptions.create(config.getPollingInterval(),
+                () -> Schedulers.newSingle(AloPollingReceiver.class.getSimpleName()));
+    }
+
+    private Flux<Alo<PolledRecord<P,O>>> createAloRecords(final Flux<ReceiverRecord<P, O>> records,
+                                          final AloFactory<PolledRecord<P, O>> aloFactory) {
+        final Sinks.Empty<Alo<PolledRecord<P, O>>> sink = Sinks.empty();
+        return records.map(record -> aloFactory.create(record.getRecord(),
+                        () -> record.getPollable().ack(record.getRecord().getOffset()),
+                        t -> {
+                            sink.tryEmitError(t);
+                            record.getPollable().nack(t, record.getRecord().getOffset());
```

Comment:
I would recommend reversing the order of these two. It could possibly be the case that emitting the error synchronously causes the `Pollable` to be closed, which would make the nack not work.

It may also be the case that you only want to do one of these two options. For instance, if the `Pollable` implements nack, that may be considered a successfully processed message if an alternate nack execution succeeds

Reply:
Example in the RabbitMQ Receiver: https://github.com/atleon/atleon/blob/main/rabbitmq/src/main/java/io/atleon/rabbitmq/AloRabbitMQReceiver.java#L166-L178